### ISSUE TITLE
[Story] Ensure the `<Box aria-label>` is flagged as inaccessible markup

### ISSUE THREAD

**Issue by khiga8** [primer/eslint-plugin-primer-react#170]
## Context

I noticed there is a gap in flagging `aria-label`/`aria-labelledby` misuse on the `Box` component!

I noticed that the following markup will be flagged by the [a11y-role-supports-aria-props](https://github.com/github/eslint-plugin-github/blob/main/docs/rules/a11y-role-supports-aria-props.md) rule:

```.tsx
<Box as="div" aria-label></Box>
<Box as="span" aria-labelledby></Box>
```

This is because we have the `polymorphicPropName` config set to `as`, so the linter knows how to map the component to an element.

However, the following markup will not be flagged:
```.tsx
<Box aria-label></Box>
<Box aria-labelledby></Box>
```

It looks like we can expand coverage and ensure this `Box` is interpreted as a `div` by mapping `Box` to `div` in the component mapping for `github`!

## Acceptance criteria

* [ ] Ensure that invalid ARIA markup on `Box` is flagged.

**Comment by joshblack** [comment]
Thanks for suggesting this, @khiga8! Just wanted to ask, where would be the place for us to set the default role for `Box`? 👀 Was having a hard time tracking it down today during triage.

**Comment by khiga8** [comment]
Hi @joshblack! [components.js](https://github.com/primer/eslint-plugin-primer-react/blob/main/src/configs/components.js) seems to be the appropriate file. 

I would suggest only passing this mapping of `Box` to `div` for the `github` plugin, since polymorphic component linting support for `jsx-a11y` has not been enabled yet (whereas it is supported in our `github` rules), and we'd want to be wary of the `jsx-a11y` improperly always treating `Box` as `div`.
